lib/rspec/core/hooks.rb

```diff
@@ -455,7 +455,7 @@ def register(prepend_or_append, position, *args, &block)
         def run(position, scope, example_or_group)
           return if RSpec.configuration.dry_run?
 
-          if scope == :context
+          if scope == :context && !example_or_group.class.metadata[:skip]
```

Comment:
Ah, yeah, sorry. I guess I was relying on the tests a little too heavily to do my thinking for me! When they passed I figured I was good. Fixed now!